aiida/backends/sqlalchemy/migrations/versions/1de112340b16_django_parity_1.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+# pylint: disable=invalid-name,no-member
+"""Parity with Django backend (rev: 0048),
+part 1: Ensure fields to make non-nullable are not currently null
+
+Revision ID: 2b40c8131fe0
+Revises: 535039300e4a
+Create Date: 2021-08-24 18:52:45.882712
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import JSONB, UUID
+
+from aiida.common import timezone
+from aiida.common.utils import get_new_uuid
+
+# revision identifiers, used by Alembic.
+revision = '1de112340b16'
+down_revision = '34a831f4286d'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():  # pylint: disable=too-many-statements
+    """Convert null values to default values"""
+    db_dbauthinfo = sa.sql.table(
+        'db_dbauthinfo',
+        # sa.Column('aiidauser_id', sa.Integer),
+        # sa.Column('dbcomputer_id', sa.Integer),
+        sa.Column('enabled', sa.Boolean),
+        sa.Column('auth_params', JSONB),
+        sa.Column('metadata', JSONB),
+    )
+
+    op.execute(db_dbauthinfo.update().where(db_dbauthinfo.c.enabled.is_(None)).values(enabled=True))
+    op.execute(db_dbauthinfo.update().where(db_dbauthinfo.c.auth_params.is_(None)).values(auth_params={}))
+    op.execute(db_dbauthinfo.update().where(db_dbauthinfo.c.metadata.is_(None)).values(metadata={}))
+    op.execute(db_dbauthinfo.update().where(db_dbauthinfo.c.auth_params == JSONB.NULL).values(auth_params={}))
+    op.execute(db_dbauthinfo.update().where(db_dbauthinfo.c.metadata == JSONB.NULL).values(metadata={}))
+
+    db_dbcomment = sa.sql.table(
+        'db_dbcomment',
+        # sa.Column('dbnode_id', sa.Integer),
+        # sa.Column('user_id', sa.Integer),
+        sa.Column('content', sa.Text),
+        sa.Column('ctime', sa.DateTime(timezone=True)),
+        sa.Column('mtime', sa.DateTime(timezone=True)),
+        sa.Column('uuid', UUID(as_uuid=True)),
+    )
+
+    op.execute(db_dbcomment.update().where(db_dbcomment.c.content.is_(None)).values(content=''))
+    op.execute(db_dbcomment.update().where(db_dbcomment.c.mtime.is_(None)).values(mtime=timezone.now()))
+    op.execute(db_dbcomment.update().where(db_dbcomment.c.ctime.is_(None)).values(ctime=timezone.now()))
```

Comment:
This would of course really set the wrong time, but I guess that is as opposed to not having a time at all. Also don't see another solution, but it does raise the question: are there really databases out there that have nulls for these fields? Seems like that would have been a source of bugs. Since `Comment` is not used a lot, it probably never surfaced, but for the other tables maybe

Reply:
>  are there really databases out there that have nulls for these fields? 

I very much doubt it, this is really just an extra pre-caution, to ensure these migrations can actually be run